File: UnitC/ch06/example-6.27.jl
#= 
  lock5stat page 494
=#

## 1. load package
include("../../utils.jl")
using UnicodePlots
## 2. load data 
desc=Lock5Table(494,"Smiles","example-6.27",["Leniency", "Group"])
gdf=@pipe load_csv(desc.name)|>groupby(_,desc.feature[2])

## 3. desc table
data=[i.Leniency for i in gdf]
Group=names(gdf)
summary_df=DataFrame(Group=Group,
                     n=size.(data,1),
                     Mean=mean.(data),
                     Stddev=std.(data)
                     )
#@pt summary_df
#= 
┌─────────┬───────┬─────────┬─────────┐
│   Group │     n │    Mean │  Stddev │
│  String │ Int64 │ Float64 │ Float64 │
├─────────┼───────┼─────────┼─────────┤
│   smile │    34 │ 4.91176 │ 1.68087 │
│ neutral │    34 │ 4.11765 │ 1.52285 │
└─────────┴───────┴─────────┴─────────┘
=#

## 4. box plot
UnicodePlots.boxplot(Group,data,title="Grouped Boxplot", xlabel="Leniency",ylabel="Group")

"""
                 Grouped Boxplot              
                 ┌                                        ┐ 
                    ╷     ┌──────┬─────┐                 ╷  
           smile    ├─────┤      │     ├─────────────────┤  
   Group            ╵     └──────┴─────┘                 ╵  
                  ╷    ┌────┬────┐                 ╷        
         neutral  ├────┤    │    ├─────────────────┤        
                  ╵    └────┴────┘                 ╵        
                 └                                        ┘ 
                  2                 5.5                  9  
                                  Leniency                
"""

EqualVarianceTTest(data...)
#= 
Two sample t-test (equal variance)
----------------------------------
Population details:
    parameter of interest:   Mean difference
    value under h_0:         0
    point estimate:          0.794118
    95% confidence interval: (0.01749, 1.571)

Test summary:
    outcome with 95% confidence: reject h_0
    two-sided p-value:           0.0452

Details:
    number of observations:   [34,34]
    t-statistic:              2.04153849287233
    degrees of freedom:       66
    empirical standard error: 0.38898000200894833
=#

File: marimo/ch06/example-6.27.py
import marimo

__generated_with = "0.24.0"
app = marimo.App(width="medium")


@app.cell
def _():
    import marimo as mo
    import sys
    from pathlib import Path
    import numpy as np
    import polars as pl
    import matplotlib.pyplot as plt
    from scipy import stats
    from great_tables import GT
    _p = Path(mo.notebook_location()).resolve().parent
    while _p != _p.parent and not (_p / "lock5.py").exists():
        _p = _p.parent
    if str(_p) not in sys.path:
        sys.path.insert(0, str(_p))
    import lock5
    return mo, sys, Path, np, pl, plt, stats, GT, lock5


@app.cell
def _(mo):
    mo.md(r"""# Example 6.27 — Smiles and leniency (two-sample t-test)

Port of `UnitC/ch06/example-6.27.jl`: compare `Leniency` between the `smile` and
`neutral` groups, then run an equal-variance two-sample t-test.""")
    return


@app.cell
def _(lock5):
    desc = lock5.Lock5Table(494, "Smiles", "example-6.27", ["Leniency", "Group"])
    df = lock5.load_csv(desc.name)
    group, data, summary = lock5.group_summary(df, desc.feature[1], desc.feature[0])
    return desc, df, group, data, summary


@app.cell
def _(GT, mo, summary):
    mo.Html(GT(summary).tab_header(title="Leniency by group").fmt_number(columns=["Mean", "Stddev"], decimals=4).as_raw_html())
    return


@app.cell
def _(data, group, lock5):
    fig, ax = lock5.grouped_boxplot(group, data, title="Grouped Boxplot", xlabel="Leniency", ylabel="Group")
    fig
    return fig, ax


@app.cell
def _(data, lock5, mo):
    res = lock5.equal_var_ttest(data[0], data[1])
    mo.Html(lock5.gt_ttest(res).as_raw_html())
    return res


if __name__ == "__main__":
    app.run()
